File: 938.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        Stack<TreeNode> stack = new Stack<>();
        int res = 0;
        stack.push(root);
        while (stack.size()!=0){
            TreeNode node = stack.pop();
            if (node.val>=low && node.val<=high){
                res += node.val;
            }
            if(node.left != null){stack.push(node.left);}
            if(node.right != null){stack.push(node.right);}
        }
        return res;
    }
}

File: 938.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        # level order trerverse
        dq = deque()
        dq.append(root)
        sum = 0
        while(dq):
            node = dq.popleft()
            if low <= node.val <= high:
                sum += node.val
            if node.left:
                dq.append(node.left)
            if node.right:
                dq.append(node.right)
        return sum
        